src/models/base.model.js

```diff
@@ -19,12 +22,30 @@ module.exports = class BaseModel {
   }
 
   isNew () {
-    let result
-    if (this.id) {
-      result = false
-    } else {
-      result = true
+    return !this.id
+  }
+
+  async save (authToken, dataObject) {
+    const dynamicsDal = new DynamicsDalService(authToken)
+    try {
+      let query
+      if (this.isNew()) {
+        // New Entity
+        query = this.entity
+        this.id = await dynamicsDal.create(query, dataObject)
+      } else {
+        // Update Entity
+        query = `${this.entity}(${this.id})`
+        await dynamicsDal.update(query, dataObject)
+
+        // The following "untilComplete" can be removed when the update is successful only when the update in dynamics has fully completed.
+        if (this.untilComplete) {
```

Comment:
The untilComplete method is defined in multiple places, would be better to make it part of the base class and have it turned on via a flag if required.

Reply:
It is in Application and ConfirmRules models at the moment.